spec/recommendations-v1.json

```diff
@@ -0,0 +1,77 @@
+{
+  "$id": "/schemas/recommendations_request",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "description": "Schema for recommendations API requests received by RayServe",
+  "additionalProperties": false,
+  "definitions": {
+    "language": {
+      "type": "string",
+      "description": "The language code of the topic or channel",
+      "pattern": "^[a-z]{3}(?:-[A-Z]{2})?$"
+    },
+    "topic": {
+      "type": "object",
+      "description": "A topic in the tree structure for which to generate recommendations",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the topic content node on Studio"
+        },
+        "title": {
+          "type": "string",
+          "description": "The title of the topic"
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the topic"
+        },
+        "language": {
+          "$ref": "#/definitions/language"
+        },
+        "ancestors": {
+          "type": "array",
+          "description": "The ancestors of the channel",
+          "items": {
+            "$ref": "#/definitions/topic"
+          }
+        }
+      },
+      "required": ["id", "title", "description", "language"]
+    }
+  },
+  "properties": {
+    "target": {
+      "type": "string",
+      "description": "The target topic of the channel"
+    },
+    "ancestors": {
+      "type": "array",
+      "description": "The ancestors of the channel's target topic",
+      "items": {
+        "$ref": "#/definitions/topic"
+      }
+    },
```

Comment:
We want the embedding endpoint to support:
- A list of items to embed rather than just a single item (this will reduce overhead for bulk embedding, as we can run a larger batch through, and do fewer API calls)
- Both topics and resources

One way to handle this could be having two top level properties like this:
```
    "topics": {
      "type": "array",
      "description": "A list of topics to embed",
      "items": {
        "$ref": "#/definitions/topic"
      }
    },
    "resources": {
      "type": "array",
      "description": "A list of content resources to embed",
      "items": {
        "$ref": "#/definitions/resource"
      }
    },
```

(we then don't need to have "target" or "ancestors" here, as the topics in the list _are_ the targets, and the ancestors are contained under each of the topic items themselves)